openmdao/core/problem.py

```diff
@@ -2009,19 +2017,19 @@ def _get_implicit_connections(self):
             if a a promoted variable name matches multiple unknowns
         """
 
-        connections = {}
-        dangling = {}
+        connections = OrderedDict() #{}
+        dangling = OrderedDict() #{}
 
         abs_unames = self.root._sysdata.to_abs_uname
-
-        for prom_name, pabs_list in iteritems(self.root._sysdata.to_abs_pnames):
-            if prom_name in abs_unames:  # param has a src in unknowns
-                uprom = abs_unames[prom_name]
-                for pabs in pabs_list:
-                    connections[pabs] = ((uprom, None),)
+        
+        for prom_name, pabs_list in iteritems(self.root._sysdata.to_abs_pnames):            
+            if prom_name in abs_unames:  # param has a src in unknowns                
+                uprom = abs_unames[prom_name]                
+                for pabs in pabs_list:                    
+                    connections[pabs] = ((uprom, None),)                    
```

Comment:
can you set your editor so it doesn't introduce all the extra spaces?  I don't think there's a change here...